### ISSUE TITLE
What does CV do?

### ISSUE THREAD

**Issue by zfrenchee** [dmlc/xgboost#786]
I'm having an issue where `CV` dramatically outperforms my trained model. I'm using the python package. 

If I take a matrix and pass it to `CV`, I'm told there's less than 2% test error. If I divide up the data set on my own and run `train` I get 27% test error. `CV` isn't particularly well documented, and I'm wondering what exactly that function does. I've asked [my question on stackoverflow](http://stackoverflow.com/questions/35052629/why-does-xgboost-cross-validation-perform-so-well-while-train-predict-performs-s).

**Comment by terrytangyuan** [comment]
My questions:
1. How did you split out your hold-out dataset? 
2. Is it highly imbalanced class? If so, accuracy is probably not the appropriate metric to use.

**Comment by zfrenchee** [comment]
The split is not even: about 1/4 of the training examples are positive class, 3/4 negative examples. 
In order to create the hold out dataset I pulled evenly from positives and negatives, cutting each in half, meaning both the training set and the holdout set are the same size, and both are 1/4 positive, examples, 3/4 negative examples.

**Comment by zfrenchee** [comment]
Turns out I was being a goof. 
I was creating the training and hold out sets separately, so they had different indices for different tokens, meaning it's a miracle it was doing so much better than random chance. I think that's what confused me -- it did so much better than 50% accuracy even with totally different indexing of features.
